vnext/Microsoft.ReactNative.SharedManaged/JSValue.cs

```diff
@@ -78,6 +94,96 @@ public JSValue(double value)
     public long Int64 => m_simpleValue.Int64Value;
     public double Double => m_simpleValue.DoubleValue;
 
+    public static explicit operator string(JSValue value)
+    {
+      switch (value.Type)
+      {
+        case JSValueType.String: return value.String;
+        case JSValueType.Boolean: return value.Boolean ? "true" : "false"; 
+        case JSValueType.Int64: return value.Int64.ToString(); 
+        case JSValueType.Double: return value.Double.ToString();
+        default: return "";
+      }
+    }
+
+    public static explicit operator bool(JSValue value)
+    {
+      switch (value.Type)
+      {
+        case JSValueType.Object: return true;
+        case JSValueType.Array: return true;
+        case JSValueType.String: return !string.IsNullOrEmpty(value.String);
+        case JSValueType.Boolean: return value.Boolean;
+        case JSValueType.Int64: return value.Int64 != 0;
+        case JSValueType.Double: return value.Double != 0;
+        default: return false;
+      }
+    }
+
+    public static explicit operator sbyte(JSValue value)
+    {
+      return (sbyte)(long)value;
+    }
+
+    public static explicit operator short(JSValue value)
+    {
+      return (short)(long)value;
+    }
+
+    public static explicit operator int(JSValue value)
+    {
+      return (int)(long)value;
+    }
+
+    public static explicit operator long(JSValue value)
+    {
+      switch (value.Type)
+      {
+        case JSValueType.String: return long.TryParse(value.String, out long result) ? result : 0;
```

Comment:
>0 [](start = 96, length = 1)

(nit) that doesn't look right that we just return 0 when there was an error

Reply:
It is an explicit cast operation. Do you think we must throw here?

---
In reply to: [381560198](https://github.com/microsoft/react-native-windows/pull/4124#discussion_r381560198) [](ancestors = 381560198)